src/AssignmentModel.coffee

```diff
@@ -0,0 +1,59 @@
+_ = require 'lodash'
+formUtils = require './formUtils'
+uuid = require 'node-uuid'
+
+# Model of an assignment object that allows manipulation
+# Options are:
+# assignment: assignment object. Required
+# form: form object. Required
+# user: current username. Required
+# groups: group names of user
+module.exports = class AssignmentModel
+  constructor: (options) ->
+    @assignment = options.assignment
+    @form = options.form
+    @user = options.user
+    @username = options.username
+    @groups = options.groups or []
+    @fixRoles()
+
+  # Fixes roles to reflect status and approved fields
+  fixRoles: ->
+    # Determine deployment
+    deployment = _.findWhere(@form.deployments, { _id: @assignment.deployment })
+    if not deployment
+      throw new Error("No matching deployments for #{@form._id} user #{@username}")
```

Comment:
This is a catastrophic error. Let's remove username altogether from constructor and just use user.